### ISSUE TITLE
Failed to list *v1beta1.AddressPool”: Internal error occurred: error resolving resource

### ISSUE THREAD

**Issue by cyclinder** [metallb/metallb#1521]
Hi teams!

I have a problem with `metallb`, I installed `metallb` via helm,  and then `metallb` is running.
But I look up the log of controller and speaker , Both reports “failed to list *v1beta1.AddressPool”：

```
root@172-17-8-110:~/cyclinder/metallb/charts/metallb# kubectl logs -f metallb-controller-c7999547b-h6lxn

W0719 07:30:20.019807       1 reflector.go:324] pkg/mod/k8s.io/client-go@v0.24.0/tools/cache/reflector.go:167: failed to list *v1beta1.AddressPool: Internal error occurred: error resolving resource
E0719 07:30:20.019865       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.24.0/tools/cache/reflector.go:167: Failed to watch *v1beta1.AddressPool: failed to list *v1beta1.AddressPool: Internal error occurred: error resolving resource
W0719 07:30:30.648746       1 reflector.go:324] pkg/mod/k8s.io/client-go@v0.24.0/tools/cache/reflector.go:167: failed to list *v1beta1.AddressPool: Internal error occurred: error resolving resource
E0719 07:30:30.648815       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.24.0/tools/cache/reflector.go:167: Failed to watch *v1beta1.AddressPool: failed to list *v1beta1.AddressPool: Internal error occurred: error resolving resource
W0719 07:30:45.334153       1 reflector.go:324] pkg/mod/k8s.io/client-go@v0.24.0/tools/cache/reflector.go:167: failed to list *v1beta1.AddressPool: Internal error occurred: error resolving resource
E0719 07:30:45.334230       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.24.0/tools/cache/reflector.go:167: Failed to watch *v1beta1.AddressPool: failed to list *v1beta1.AddressPool: Internal error occurred: error resolving resource
W0719 07:31:34.325453       1 reflector.go:324] pkg/mod/k8s.io/client-go@v0.24.0/tools/cache/reflector.go:167: failed to list *v1beta1.AddressPool: Internal error occurred: error resolving resource
E0719 07:31:34.325509       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.24.0/tools/cache/reflector.go:167: Failed to watch *v1beta1.AddressPool: failed to list *v1beta1.AddressPool: Internal error occurred: error resolving resource
```
*speaker*:

```
root@172-17-8-110:~/cyclinder/metallb/charts/metallb# kubectl logs -f metallb-speaker-shn82

W0719 07:33:43.652572       1 reflector.go:324] pkg/mod/k8s.io/client-go@v0.24.0/tools/cache/reflector.go:167: failed to list *v1beta2.BGPPeer: Internal error occurred: error resolving resource
E0719 07:33:43.652627       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.24.0/tools/cache/reflector.go:167: Failed to watch *v1beta2.BGPPeer: failed to list *v1beta2.BGPPeer: Internal error occurred: error resolving resource
W0719 07:33:48.528408       1 reflector.go:324] pkg/mod/k8s.io/client-go@v0.24.0/tools/cache/reflector.go:167: failed to list *v1beta1.AddressPool: Internal error occurred: error resolving resource
```

And I look up the `AddressPoo`l  via `kubectl`:

```
root@172-17-8-110:~/cyclinder/metallb/charts/metallb# kubectl get addresspools.metallb.io
Error from server (InternalError): Internal error occurred: error resolving resource
root@172-17-8-110:~/cyclinder/metallb/charts/metallb# kubectl get bgpadvertisements.metallb.io
No resources found in default namespace.
```

After every period of time metallb will restart, after four restarts, the state of pod will change to CrashLoopBackOff:

```
root@172-17-8-110:~/cyclinder/metallb/charts/metallb# kubectl get po
NAME                                  READY   STATUS             RESTARTS      AGE
metallb-controller-c7999547b-h6lxn    0/1     CrashLoopBackOff   4 (31s ago)   12m
metallb-speaker-cwxxv                 0/1     CrashLoopBackOff   4 (28s ago)   12m
metallb-speaker-shn82                 0/1     CrashLoopBackOff   4 (34s ago)   12m
```

I found `apiserver` endlessly complaining about "unable to load root certificates: unable to parse bytes as PEM block"：

```
E0719 07:44:08.181223       1 customresource_handler.go:309] unable to load root certificates: unable to parse bytes as PEM block
E0719 07:44:10.582137       1 customresource_handler.go:309] unable to load root certificates: unable to parse bytes as PEM block
E0719 07:44:13.047572       1 crd_finalizer.go:299] addresspools.metallb.io failed with: unable to find a custom resource client for addresspools.metallb.io: unable to load root certificates: unable to parse bytes as PEM block
E0719 07:44:13.110443       1 crd_finalizer.go:299] bgppeers.metallb.io failed with: unable to find a custom resource client for bgppeers.metallb.io: unable to load root certificates: unable to parse bytes as PEM block
E0719 07:44:15.850811       1 customresource_handler.go:309] unable to load root certificates: unable to parse bytes as PEM block
E0719 07:44:24.932742       1 customresource_handler.go:309] unable to load root certificates: unable to parse bytes as PEM block
E0719 07:44:27.694629       1 customresource_handler.go:309] unable to load root certificates: unable to parse bytes as PEM block
E0719 07:44:28.447875       1 customresource_handler.go:309] unable to load root certificates: unable to parse bytes as PEM block
E0719 07:44:28.495238       1 customresource_handler.go:309] unable to load root certificates: unable to parse bytes as PEM block
```
I have no idea about this, Can anyone give me some advice? Thanks!

```
Metallb: v0.13.3
Kubernetes: v1.24.2
CNI: calico
Kube-proxy: iptables
```

**Comment by fedepaol** [comment]
I have the suspect that the apiserver doesn't like the fake ca we are adding here https://github.com/metallb/metallb/blob/6c1bd7d2b1d6ceeedbc4cb162bb7e24ca1e50244/charts/metallb/charts/crds/templates/crds.yaml#L23

Seems similar to what we had in https://github.com/metallb/metallb/issues/1501 (although there it stabilizes if we let the controller run). 
Would you mind coming on the #metallb slack channel on the kubernetes slack workspace so we can triage this?

If not, I'd like to have the output of:
`kubectl get crds -A | grep metallb`

if addresspool is listed, then the output of
`kubectl get crds addresspools.metallb.io -o yaml`

If it's not, try to apply the following yaml, with your namespace instead of `{{ .Release.Namespace }}` :

```
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  annotations:
    controller-gen.kubebuilder.io/version: v0.7.0
  creationTimestamp: null
  name: addresspools.metallb.io
spec:
  group: metallb.io
  names:
    kind: AddressPool
    listKind: AddressPoolList
    plural: addresspools
    singular: addresspool
  scope: Namespaced
  conversion:
    strategy: Webhook
    webhook:
      conversionReviewVersions: ["v1alpha1", "v1beta1"]
      clientConfig:
        # this is "\n" used as a placeholder, otherwise it will be rejected by the apiserver for being blank,
        # but we're going to set it later using the cert-controller.
        caBundle: Cg==
        service:
          namespace: {{ .Release.Namespace }}
          name: metallb-webhook-service
          path: /convert
  versions:
  - deprecated: true
    deprecationWarning: metallb.io v1alpha1 AddressPool is deprecated
    name: v1alpha1
    schema:
      openAPIV3Schema:
        description: AddressPool is the Schema for the addresspools API.
        properties:
          apiVersion:
            description: 'APIVersion defines the versioned schema of this representation
              of an object. Servers should convert recognized schemas to the latest
              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
            type: string
          kind:
            description: 'Kind is a string value representing the REST resource this
              object represents. Servers may infer this from the endpoint the client
              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
            type: string
          metadata:
            type: object
          spec:
            description: AddressPoolSpec defines the desired state of AddressPool.
            properties:
              addresses:
                description: A list of IP address ranges over which MetalLB has authority.
                  You can list multiple ranges in a single pool, they will all share
                  the same settings. Each range can be either a CIDR prefix, or an
                  explicit start-end range of IPs.
                items:
                  type: string
                type: array
              autoAssign:
                default: true
                description: AutoAssign flag used to prevent MetallB from automatic
                  allocation for a pool.
                type: boolean
              bgpAdvertisements:
                description: When an IP is allocated from this pool, how should it
                  be translated into BGP announcements?
                items:
                  properties:
                    aggregationLength:
                      default: 32
                      description: The aggregation-length advertisement option lets
                        you “roll up” the /32s into a larger prefix.
                      format: int32
                      minimum: 1
                      type: integer
                    aggregationLengthV6:
                      default: 128
                      description: Optional, defaults to 128 (i.e. no aggregation)
                        if not specified.
                      format: int32
                      type: integer
                    communities:
                      description: BGP communities
                      items:
                        type: string
                      type: array
                    localPref:
                      description: BGP LOCAL_PREF attribute which is used by BGP best
                        path algorithm, Path with higher localpref is preferred over
                        one with lower localpref.
                      format: int32
                      type: integer
                  type: object
                type: array
              protocol:
                description: Protocol can be used to select how the announcement is
                  done.
                enum:
                - layer2
                - bgp
                type: string
            required:
            - addresses
            - protocol
            type: object
          status:
            description: AddressPoolStatus defines the observed state of AddressPool.
            type: object
        required:
        - spec
        type: object
    served: true
    storage: false
    subresources:
      status: {}
  - deprecated: true
    deprecationWarning: metallb.io v1beta1 AddressPool is deprecated, consider using
      IPAddressPool
    name: v1beta1
    schema:
      openAPIV3Schema:
        description: AddressPool represents a pool of IP addresses that can be allocated
          to LoadBalancer services. AddressPool is deprecated and being replaced by
          IPAddressPool.
        properties:
          apiVersion:
            description: 'APIVersion defines the versioned schema of this representation
              of an object. Servers should convert recognized schemas to the latest
              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
            type: string
          kind:
            description: 'Kind is a string value representing the REST resource this
              object represents. Servers may infer this from the endpoint the client
              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
            type: string
          metadata:
            type: object
          spec:
            description: AddressPoolSpec defines the desired state of AddressPool.
            properties:
              addresses:
                description: A list of IP address ranges over which MetalLB has authority.
                  You can list multiple ranges in a single pool, they will all share
                  the same settings. Each range can be either a CIDR prefix, or an
                  explicit start-end range of IPs.
                items:
                  type: string
                type: array
              autoAssign:
                default: true
                description: AutoAssign flag used to prevent MetallB from automatic
                  allocation for a pool.
                type: boolean
              bgpAdvertisements:
                description: Drives how an IP allocated from this pool should translated
                  into BGP announcements.
                items:
                  properties:
                    aggregationLength:
                      default: 32
                      description: The aggregation-length advertisement option lets
                        you “roll up” the /32s into a larger prefix.
                      format: int32
                      minimum: 1
                      type: integer
                    aggregationLengthV6:
                      default: 128
                      description: Optional, defaults to 128 (i.e. no aggregation)
                        if not specified.
                      format: int32
                      type: integer
                    communities:
                      description: BGP communities to be associated with the given
                        advertisement.
                      items:
                        type: string
                      type: array
                    localPref:
                      description: BGP LOCAL_PREF attribute which is used by BGP best
                        path algorithm, Path with higher localpref is preferred over
                        one with lower localpref.
                      format: int32
                      type: integer
                  type: object
                type: array
              protocol:
                description: Protocol can be used to select how the announcement is
                  done.
                enum:
                - layer2
                - bgp
                type: string
            required:
            - addresses
            - protocol
            type: object
          status:
            description: AddressPoolStatus defines the observed state of AddressPool.
            type: object
        required:
        - spec
        type: object
    served: true
    storage: true
    subresources:
      status: {}
status:
  acceptedNames:
    kind: ""
    plural: ""
  conditions: []
  storedVersions: []
```

**Comment by mohahmed13** [comment]
Just in case someone runs across this in the future, the issue here (and the subsequent fix in v0.13.4) resulted in Argo CD in getting a cluster connection and sync failure: `failed to sync cluster [https://10.43.0.1:443](https://10.43.0.1/): failed to load initial state of resource [BGPPeer.metallb.io](http://bgppeer.metallb.io/): Internal error occurred: error resolving resource`. This appeared to happen when our K8s version went to v1.23. 

This caused all our apps to go into an "Unknown" state and generally weird behaviour. I'm unsure if it was isolated to Argo CD or if there were other implications. Updating metallb to the version which had this fix resolved the issue.
